### PR TITLE
test(docs-infra): reduce CI flakiness

### LINKED ISSUE #29544: ci: `lifecycle-hooks` docs example e2e tests are flakey under Ivy
The `lifecycle-hooks` docs example e2e tests (which are part of the [test_docs_examples_ivy][1] CircleCI job) are flakey under Ivy.
Examples failures:

Job [260998][2]
```
Lifecycle hooks
  ✗ should support AfterContent hooks
    - Expected 9 to equal 10.
        at /home/circleci/ng/aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts:136:28
        at ManagedPromise.invokeCallback_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:1376:14)
        at TaskQueue.execute_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:3084:14)
        at TaskQueue.executeNext_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:3067:27)
        at asyncRun (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:2927:27)
        at /home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:668:7
        at process._tickCallback (internal/process/next_tick.js:68:7)
```

Job [261278][3]
```
Lifecycle hooks
  ✗ should support DoCheck hook
    - Expected 9 to equal 8.
        at /home/circleci/ng/aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts:75:21
        at ManagedPromise.invokeCallback_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:1376:14)
        at TaskQueue.execute_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:3084:14)
        at TaskQueue.executeNext_ (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:3067:27)
        at asyncRun (/home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:2927:27)
        at /home/circleci/ng/aio/tools/examples/shared/node_modules/selenium-webdriver/lib/promise.js:668:7
        at process._tickCallback (internal/process/next_tick.js:68:7)
```

[1]: https://github.com/angular/angular/blob/a28b3e33598cf6a9fae685d8c1a1e39b419d1c78/.circleci/config.yml#L339
[2]: https://circleci.com/gh/angular/angular/260998
[3]: https://circleci.com/gh/angular/angular/261278

### PR DESCRIPTION
**Pull request by gkalpak**
Refactor specific docs examples e2e tests to reduce flakiness on CI.
(See individual commits for more details.)

Fixes #29544.

### CODE REVIEW

File: aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts [hunk 1]
```diff
@@ -2,30 +2,31 @@
...
     expect(element(by.css('h1')).getText()).toEqual(_title);
   });
 
-  it('should be able to select green highlight', function () {
-    let highlightedEle = element(by.cssContainingText('p', 'Highlight me!'));
-    let lightGreen = 'rgba(144, 238, 144, 1)';
+  it('should be able to select green highlight', () => {
+    const highlightedEle = element(by.cssContainingText('p', 'Highlight me!'));
+    const lightGreen = 'rgba(144, 238, 144, 1)';
+    const getBgColor = () => highlightedEle.getCssValue('background-color');
```
Comment (petebacondarwin) [inline aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts]: `getBgColor` doesn't appear to be used?

File: aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts [hunk 2]
```diff
@@ -2,30 +2,31 @@
...
-    let greenRb = element.all(by.css('input')).get(0);
-    greenRb.click().then(function() {
-      // TypeScript Todo: find the right type for highlightedEle
-      browser.actions().mouseMove(highlightedEle as any).perform();
-      expect(highlightedEle.getCssValue('background-color')).toEqual(lightGreen);
-    });
 
+    const greenRb = element.all(by.css('input')).get(0);
+    greenRb.click();
+    browser.actions().mouseMove(highlightedEle).perform();
```
Comment (petebacondarwin) [inline aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts]: Does `.perform()` cause this to happen synchronously?

File: aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts [hunk 3]
```diff
@@ -1,39 +1,42 @@
...
-      expect(updateHeroButton.isPresent()).toBe(true, 'should be able to see the update hero button');
-      return updateHeroButton.click();
-    }).then(function () {
-      expect(pabComp.getText()).toContain('Windstorm!');
-      return pabButton.click();
-    }).then(function () {
-      expect(pabComp.isPresent()).toBe(false, 'should no longer be able to find the "peek-a-boo" component');
-    });
+
+    await pabButton.click();
```
Comment (petebacondarwin) [inline aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts]: No `.perform()` here?

**Review by petebacondarwin — APPROVED** [review]
A nit or two but otherwise LGTM

File: aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts (hunk 1 above)
Comment (gkalpak) [reply aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts]: Ooops. It introduced it, then decided it is not worth it, but forgot to remove the declaration 😇

File: aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts (hunk 2 above)
Comment (gkalpak) [reply aio/content/examples/attribute-directives/e2e/src/app.e2e-spec.ts]: No, (although it shouldn't matter here). It just how WebDriver's [browser.actions()][1] needs to be used.

[1]: https://www.protractortest.org/#/api?view=webdriver.WebDriver.prototype.actions

File: aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts (hunk 3 above)
Comment (gkalpak) [reply aio/content/examples/lifecycle-hooks/e2e/src/app.e2e-spec.ts]: It is only for WebDriver's [browser.actions()][1]. [.click()][2] is a WebElement method.

[1]: https://www.protractortest.org/#/api?view=webdriver.WebDriver.prototype.actions
[2]: https://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.click

**Pull request closed without merging**